### DISCUSSION TITLE
Lots of Pylance type hint errors

### DISCUSSION THREAD

**Discussion by yanxke** [isl-org/Open3D#6234] [General]
For example

```
import open3d.visualization
from open3d.visualization import Visualizer

vis = open3d.visualization.Visualizer()
```
yields "Visualizer" is not a known member of module "open3d.visualization" and "Visualizer" is unknown import symbol.

How should we fix this?

**Comment by errissa** [comment]
Unfortunately, since we don't use type annotations or generate type stubs Pylance and other Python language servers have limited ability to understand Open3D. You could try automatically generating type stubs with a tool like `stubgen` and I believe pyright has an option to generate stubs as well. I've never tried it myself so I don't know how effective auto-generated stubs would be for Open3D.
